framework/src/meshlet_helpers.cpp

```diff
@@ -0,0 +1,139 @@
+		for (auto& pair : aModels) {
+			auto model = std::get<avk::resource_ownership<model_t>>(pair);
+			auto& meshIndices = std::get<std::vector<mesh_index_t>>(pair);
+
+			if (aCombineSubmeshes) {
+				auto [vertices, indices] = get_vertices_and_indices(std::vector({ std::make_tuple(avk::const_referenced(model.get()), meshIndices) }));
+				// default to very bad meshlet generation
+				std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, model, std::nullopt, aMaxVertices, aMaxIndices, aMeshletDivision);
+				// append to meshlets
+				meshlets.insert(std::end(meshlets), std::begin(tmpMeshlets), std::end(tmpMeshlets));
+			}
+			else {
+				for (const auto meshIndex : meshIndices) {
+					auto vertices = model.get().positions_for_mesh(meshIndex);
+					auto indices = model.get().indices_for_mesh<uint32_t>(meshIndex);
+					// default to very bad meshlet generation
+					std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, model, meshIndex, aMaxVertices, aMaxIndices, aMeshletDivision);
+					// append to meshlets
+					meshlets.insert(std::end(meshlets), std::begin(tmpMeshlets), std::end(tmpMeshlets));
+				}
+			}
+		}
+		return meshlets;
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_into_meshlets_cached(gvk::serializer& aSerializer,
+		const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		F&& aMeshletDivision, const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		std::vector<meshlet> meshlets;
+		if (aSerializer.mode() == gvk::serializer::mode::serialize) {
+			meshlets = divide_into_meshlets(aModels, aMeshletDivision, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+		}
+		aSerializer.archive(meshlets);
+
+		if (aSerializer.mode() == serializer::mode::deserialize) {
+			std::unordered_map<std::string, gvk::model> modelLookup;
+			for (auto& pair : aModels) {
+				auto model = std::get<avk::resource_ownership<gvk::model_t>>(pair);
+				modelLookup.insert_or_assign(model->path(), model.own());
+			}
+			for (auto& meshlet : meshlets) {
+				meshlet.mModel = modelLookup.at(meshlet.mModelPath);
+			}
+		}
+
+		return meshlets;
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_vertices_into_meshlets(const std::vector<glm::vec3>& aVertices,
+		const std::vector<uint32_t>& aIndices, avk::resource_ownership<gvk::model_t> aModel,
+		const std::optional<mesh_index_t> aMeshIndex, const uint32_t aMaxVertices, const uint32_t aMaxIndices,
+		F&& aMeshletDivision)
+	{
+		std::vector<meshlet> tmpMeshlets;
+		if constexpr (std::is_assignable_v<std::function<std::vector<meshlet>(const std::vector<uint32_t>&tIndices, avk::resource_ownership<model_t> tModel, const std::optional<mesh_index_t> tMeshIndex, const uint32_t tMaxVertices, const uint32_t tMaxIndices)>, decltype(aMeshletDivision)>) {
+			tmpMeshlets = aMeshletDivision(aIndices, aModel, aMeshIndex, aMaxVertices, aMaxIndices);
```

Comment:
_WE_ should take care of taking ownership of `aModel`. I.e., I suggest the following: 
1. `aModel.own()`
2. `.enable_shared_ownership()` or the owned model
3. In a for-each loop after the invocation of `aMeshletDivision`, assign the owned and shared model to each meshlet!

_Important:_ Document that the lambda passed to the `F`-parameter should _not_ store the model in `struct model` but only populate the other members!